pandora-server-directory/src/spaces/space.ts

```diff
@@ -462,7 +506,7 @@ export class Space {
 
 	/** Returns if this space is visible to the specific account when searching in space search */
 	public checkVisibleTo(account: Account): boolean {
-		return this.isValid && (this.isAdmin(account) || this.isPublic);
+		return this.isValid && (this.isPublic || this.isAdmin(account) || this.config.allow.includes(account.id));
```

Comment:
Might be nice to add `isAllowed(account: Account)`, similar to the existing `isAdmin`/`isBanned` helpers. It would also be implied by `isAdmin`.